src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -41,35 +44,40 @@ class TaskDeploymentService {
-        val iamRole = awsCommunicator.createIamRole(correlationId, "orchestration-service-user-$userName-role", taskAssumeRoleString)
-        awsCommunicator.attachIamPolicyToRole(correlationId, iamPolicy, iamRole)
-
-        // ECS
-        val ecsServiceName = "$userName-analytical-workspace"
-//        awsCommunicator.createEcsService(correlationId, ecsClusterName, ecsServiceName, taskDefinition, ecsLoadBalancer)
-        val containerOverrides = buildContainerOverrides(correlationId, userName, emrClusterHostName, jupyterMemory, jupyterCpu)
-        val ecsTaskRequest = awsCommunicator.buildEcsTask(ecsClusterName, taskDefinition, iamRole.arn(), containerOverrides)
-
-        Thread.sleep(30000)
-        awsCommunicator.runEcsTask(correlationId, ecsTaskRequest)
-
-        return UserTask(correlationId, userName, targetGroup.targetGroupArn(), albRoutingRule.ruleArn(), ecsClusterName, ecsServiceName, iamRole.roleName(), iamPolicy.arn())
+        //Create an entry in DynamoDB for current deployment
+        activeUserTasks.initialiseDeploymentEntry(correlationId, userName)
+
+        try {
+            // Load balancer & Routing
+            val loadBalancer = awsCommunicator.getLoadBalancerByName(albName)
+            val listener = awsCommunicator.getAlbListenerByPort(loadBalancer.loadBalancerArn(), albPort)
+            val targetGroup = awsCommunicator.createTargetGroup(correlationId, userName, loadBalancer.vpcId(), containerPort)
+            // There are 2 distinct LoadBalancer classes in the AWS SDK - ELBV2 and ECS. They represent the same LB but in different ways.
+            // The following is the load balancer needed to create an ECS service.
+//            val ecsLoadBalancer = LoadBalancer.builder()
+//                    .targetGroupArn(targetGroup.targetGroupArn())
+//                    .containerName("guacamole")
+//                    .containerPort(containerPort)
+//                    .build()
+            awsCommunicator.createAlbRoutingRule(correlationId, userName, listener.listenerArn(), targetGroup.targetGroupArn())
+
+            // IAM permissions
+            parsePolicyDocuments(additionalPermissions)
+            val iamPolicy = awsCommunicator.createIamPolicy(correlationId, userName, taskRolePolicyString)
+            val iamRole = awsCommunicator.createIamRole(correlationId, userName, taskAssumeRoleString)
+            awsCommunicator.attachIamPolicyToRole(correlationId, iamPolicy, iamRole)
+
+            // ECS
+//        awsCommunicator.createEcsService(correlationId, userName, ecsClusterName, taskDefinition, ecsLoadBalancer)
+            val containerOverrides = buildContainerOverrides(correlationId, userName, emrClusterHostName, jupyterMemory, jupyterCpu)
+            val ecsTaskRequest = awsCommunicator.buildEcsTask(ecsClusterName, taskDefinition, iamRole.arn(), containerOverrides)
+
+            Thread.sleep(30000)
```

Comment:
I know this sleep is in the prefactored code, but it seems a long time to wait, will our initial /connect call to OS complete within 60 seconds?

Reply:
We are waiting because of a potential issue with IAM, which is eventually consistent and without the wait the IAM roles wouldn't propagate in time for the task to run. This issue may be avoided when moving to services instead of running the task

Reply:
Yep, as above, it was used for IAM. Hopefully #49 will remove the need for this since we are no longer deploying a standalone task